test/annotation_utils.t

```diff
@@ -864,6 +980,112 @@ subtest '_run_rast_workflow_ann' => sub {
 };
 
 
+## Just print to explore the finalgenome data structure and find the contig_ids##
+subtest '_check_contigID_mapping' => sub {
+    my $gn_pairs = [
+        {gn => $ann_genome01, rd_ref => \%rast_details01},
+        {gn => $ann_genome02, rd_ref => \%rast_details02},
+        {gn => $ann_genome1, rd_ref => \%rast_details1},
+        {gn => $ann_genome2, rd_ref => \%rast_details2}
+    ];
+    for my $p (@$gn_pairs) {
+        my %rd = %{ $p->{rd_ref} };
+        my $genome_before_remapping = $p->{gn};
+        my $ctgID_hash = $rd{contigID_hash};
+        if( $ctgID_hash ) {
+            print "\nThere is a contigID_hash:\n".Dumper($ctgID_hash);
+        }
+        else {
+            print "\nThere is NO contigID_hash in $genome_before_remapping->{id}:\n".Dumper(keys %rd);
+            next;
+        }
+
+        ## Run the remapping
+        my $genome_after_remapping = $annoutil->_remap_contigIDs( $ctgID_hash, $genome_before_remapping );
+        print "\n=========after remapping the rasted genome on $genome_before_remapping->{id}:\n";
+
+        my $exp_ctg_ids = [];
+        if( $genome_after_remapping->{contig_ids} &&
+                @{$genome_after_remapping->{contig_ids}} > 0) {
+            for my $cid (@{$genome_before_remapping->{contig_ids}}) {
+                if( $ctgID_hash->{$cid} ) {
+                    push @{$exp_ctg_ids}, $ctgID_hash->{$cid};
+                } else {
+                    push @{$exp_ctg_ids}, $cid;
+                }
+            }
+            cmp_deeply $genome_after_remapping->{contig_ids}, $exp_ctg_ids,
+                        'contig ids remapped correctly';
+        }
+
+        $exp_ctg_ids = [];
+        my $result_ctg_ids = [];
+        if( $genome_after_remapping->{contigs} &&
+                @{$genome_after_remapping->{contigs}} > 0) {
```

Comment:
`@{$genome_after_remapping->{contigs}}` will test that the array is not empty

Reply:
Obviously I don't know Perl that well.  I will change that.